### PR TITLE
Feature/register

### PR DESCRIPTION
**Pull request by Jshak**
POST /api/auth/register
POST /api/auth/login
jwt token is passed in reponse body of login
Bearer token disabled for now.

### CODE REVIEW

File: src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public ResponseEntity<String> register(@RequestBody RegisterDto register) {
+        if (userLoginRepository.existsByUsername(register.getUsername())) {
+            return new ResponseEntity<>("Error: Username is already taken", HttpStatus.BAD_REQUEST);
+        }
+
+        if (userLoginRepository.existsByEmail(register.getEmail())) {
+            return new ResponseEntity<>("Error: Email is already in use", HttpStatus.BAD_REQUEST);
+        }
+
+        UserLoginEntity user = new UserLoginEntity();
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java:48]: Minor detail tara @Builder raicha yo entity ma, tei chalauda kasto hola?

File: src/main/java/com/laundrycorporation/projectzero/dto/LoginDto.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.laundrycorporation.projectzero.dto;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+
+@Data
+@Builder
+public class LoginDto {
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/dto/LoginDto.java:9]: Dto ko satta Response/Request terms use garum

File: src/main/java/com/laundrycorporation/projectzero/dto/RegisterDto.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.laundrycorporation.projectzero.dto;
+
+
+import lombok.Builder;
+import lombok.Data;
+@Builder
+@Data
+public class RegisterDto {
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/dto/RegisterDto.java:8]: Same here.

File: src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    @JsonIgnore
+    private String password;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
+    }
+
+    @Override
+    public String getPassword() {
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java:32]: Esma Lombok ko getter setter use nagarne?

File: src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java:44]: Yo hardcode gareko ho etai? DB bata auna parne haina?

File: src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        user.setUsername(register.getUsername());
+        user.setEmail(register.getEmail());
+        user.setPassword(passwordEncoder.encode(register.getPassword()));
+
+        userLoginRepository.save(user);
```
Comment (rojan96) [inline src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java:53]: Repository siddhai controller ma call nagarera service class ni banaune ho ki? Aile lai ta no issues tara pachi code base bade pachi seperation of concerns huda ali clean huncha. So Controller ma only request in response out, aru sabai Service ma handle garne.

**Review by rojan96 — COMMENTED** [review]
Nice! Left a few small comments, but looks great overall!

File: src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java (hunk 1 above)
Comment (Jshak) [reply src/main/java/com/laundrycorporation/projectzero/controller/AuthController.java]: done

File: src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java (hunk 5 above)
Comment (Jshak) [reply src/main/java/com/laundrycorporation/projectzero/security/service/UserDetailsImpl.java:36]: roles set gare pachhy we can maintain this in db aile lai this is fine

**Comment by Jshak** [comment]
@rojan96 thanks for reviewing

**Review by bniroula — APPROVED** [review]
Looks really good dai! 👍

**Review by rojan96 — APPROVED** [review]

**Pull request merged**